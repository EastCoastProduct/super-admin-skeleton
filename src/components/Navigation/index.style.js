import { StyleSheet } from 'aphrodite/no-important';
import variables from '../../styles/variables';

const { colors, sizes } = variables;

const mixin = {
  navLink: {
    display: 'block',
    fontWeight: 700,
    lineHeight: '80px',
    padding: '0 25px',
    textDecoration: 'none',
    textTransform: 'uppercase',
  },
};

export default StyleSheet.create({
  aside: {
    backgroundColor: colors.contentBcg,
    height: `calc(100vh - ${sizes.header}px)`,
    left: 0,
    overflow: 'auto',
    padding: '90px 0 20px',
    position: 'fixed',
    top: sizes.header,
    width: sizes.navigation,
  },
  navLink: {
    ...mixin.navLink,
    color: colors.header,
  },
  navLinkActive: {
    ...mixin.navLink,
    backgroundColor: colors.header,
    color: colors.contentBcg,
  },
  navIcon: {
    fontSize: 20,
    marginRight: 10,
    width: 20,
  },
});
