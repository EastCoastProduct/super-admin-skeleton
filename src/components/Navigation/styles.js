import { StyleSheet } from 'aphrodite/no-important';
import { colors, sizes, zIndex } from '../../styles/variables';

const shared = {
  navLink: {
    display: 'block',
    lineHeight: '80px',
    padding: '0 25px',
    textDecoration: 'none',
  },
};

export default StyleSheet.create({
  aside: {
    backgroundColor: colors.contentBcg,
    height: `calc(100vh - ${sizes.header}px)`,
    left: 0,
    overflow: 'auto',
    padding: '90px 0 20px',
    position: 'fixed',
    top: sizes.header,
    width: sizes.navigation,
    zIndex: zIndex.navigation,
  },
  navLink: {
    ...shared.navLink,
    color: colors.header,
  },
  navLinkActive: {
    ...shared.navLink,
    backgroundColor: colors.header,
    color: colors.contentBcg,
  },
  navIcon: {
    fontSize: 20,
    marginRight: 10,
    width: 20,
  },
});
