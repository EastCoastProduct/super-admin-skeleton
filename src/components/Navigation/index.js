import React from 'react';
import { IndexLink, Link } from 'react-router';
import { css } from 'aphrodite/no-important';
import styles from './styles';

const Navigation = () =>
  <aside className={css(styles.aside)}>
    <nav>
      <ul>
        <li>
          <IndexLink
            className={css(styles.navLink)}
            activeClassName={css(styles.navLinkActive)}
            to="/"
          >
            <i className={`${css(styles.navIcon)} fa fa-user`} />
            Users
          </IndexLink>
        </li>
        <li>
          <Link
            className={css(styles.navLink)}
            activeClassName={css(styles.navLinkActive)}
          >
            <i className={`${css(styles.navIcon)} fa fa-address-book`} />
            Placeholder
          </Link>
        </li>
        <li>
          <Link
            className={css(styles.navLink)}
            activeClassName={css(styles.navLinkActive)}
          >
            <i className={`${css(styles.navIcon)} fa fa-lock`} />
            Placeholder
          </Link>
        </li>
      </ul>
    </nav>
  </aside>;

export default Navigation;
