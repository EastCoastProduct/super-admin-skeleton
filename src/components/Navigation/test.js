import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './';

describe('Navigation component snapshot', () => {
  it('renders', () => {
    const wrapper = shallow(<Navigation />);

    expect(wrapper).toMatchSnapshot();
  });
});
