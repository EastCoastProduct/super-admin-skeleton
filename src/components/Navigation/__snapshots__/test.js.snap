exports[`Navigation component snapshot renders 1`] = `
<aside
  className="aside_r524fi">
  <nav>
    <ul>
      <li>
        <IndexLink
          activeClassName="navLinkActive_8fjitp"
          className="navLink_1dyv036"
          to="/">
          <i
            className="navIcon_11u5mqt fa fa-user" />
          Users
        </IndexLink>
      </li>
      <li>
        <Link
          activeClassName="navLinkActive_8fjitp"
          className="navLink_1dyv036"
          onlyActiveOnIndex={false}
          style={Object {}}>
          <i
            className="navIcon_11u5mqt fa fa-asterisk" />
          Placeholder
        </Link>
      </li>
      <li>
        <Link
          activeClassName="navLinkActive_8fjitp"
          className="navLink_1dyv036"
          onlyActiveOnIndex={false}
          style={Object {}}>
          <i
            className="navIcon_11u5mqt fa fa-asterisk" />
          Placeholder
        </Link>
      </li>
    </ul>
  </nav>
</aside>
`;
